lib/lte_link_control/lte_lc.c

```diff
@@ -67,12 +67,25 @@ static char edrx_param_nbiot[5] = CONFIG_LTE_EDRX_REQ_VALUE_NBIOT;
 /* Default PTW setting */
 static char ptw_param_ltem[5] = CONFIG_LTE_PTW_VALUE_LTE_M;
 static char ptw_param_nbiot[5] = CONFIG_LTE_PTW_VALUE_NBIOT;
+
+#if (CONFIG_LTE_PSM_REQ_RAT_SECONDS != -1)
 /* Default PSM RAT setting */
 static char psm_param_rat[9] = CONFIG_LTE_PSM_REQ_RAT;
-/* Default PSM RPATU setting */
+#else
+static char psm_param_rat[9] = CONFIG_LTE_PSM_REQ_RAT_SECONDS;
+#endif
+
+#if (CONFIG_LTE_PSM_REQ_RPTAU_SECONDS != -1)
+
+/* Default PSM RPTAU setting */
 static char psm_param_rptau[9] = CONFIG_LTE_PSM_REQ_RPTAU;
-/* Request PSM to be disabled */
+#else
+static char psm_param_rptau[9] = CONFIG_LTE_PSM_REQ_RPTAU_SECONDS;
+#endif
```

Comment:
This won't work at all and I have no idea how to implement those new Kconfigs nicely.

Reply:
How about adding a Kconfig choice to select if binary string or seconds are used as default values. Setting the choice to binary string as default would maintain current behavior

Reply:
I've implemented a choice symbol and code into `init_and_config()` to handle this.

Reply:
This looks like a good solution 👍